devicetypes/qubino/qubino-flush-dimmer.src/qubino-flush-dimmer.groovy

```diff
@@ -0,0 +1,890 @@
+		componentLabel = notificationComponentLabel[cmd.notificationType as int].find{ it.key == "componentLabel" }?.value
+		childDTHname = notificationComponentLabel[cmd.notificationType as int].find{ it.key == "dthName" }?.value
+
+		if (cmd.notificationType == 0x01) { // Smoke Alarm
+			map.name = "smoke"
+			switch (cmd.event) {
+				case 0x02:
+					map.value = "detected"
+					map.descriptionText = "$device.displayName detected smoke"
+					break
+				case 0x00:
+					map.value = "clear"
+					map.descriptionText = "$device.displayName detected no smoke"
+					break
+			}
+		} else if (cmd.notificationType == 0x02) { // CO Alarm
+			map.name = "carbonMonoxide"
+			switch (cmd.event) {
+				case 0x02:
+					map.value = "detected"
+					map.descriptionText = "$device.displayName detected CO"
+					break
+				case 0x00:
+					map.value = "clear"
+					map.descriptionText = "$device.displayName detected no CO"
+					break
+			}
+		} else if (cmd.notificationType == 0x03) { // C02 Alarm
+			map.name = "carbonDioxide"
+			switch (cmd.event) {
+				case 0x02:
+					map.value = "detected"
+					map.descriptionText = "$device.displayName detected CO2"
+					break
+				case 0x00:
+					map.value = "clear"
+					map.descriptionText = "$device.displayName detected no CO2"
+					break
+			}
+		} else if (cmd.notificationType == 0x04) { // Overheat Alarm
+			map.name = "temperatureAlarm"
+			switch (cmd.event) {
+				case 0x00:
+					map.value = "cleared"
+					map.descriptionText = "$device.displayName heat is clear"
+					break
+				case 0x02:
+					map.value = "heat"
+					map.descriptionText = "$device.displayName heat was detected!"
+					break
+			}
+		} else if (cmd.notificationType == 0x05) { // Water Alarm
+			map.name = "water"
+			switch (cmd.event) {
+				case 0x02:
+					map.value = "wet"
+					map.descriptionText = "$device.displayName detected water"
+					break
+				case 0x00:
+					map.value = "dry"
+					map.descriptionText = "$device.displayName detected no water"
+					break
+			}
+		} else if (cmd.notificationType == 0x07) { // Home Security
+			map.name = "motion"
+			if (cmd.event == 0x00) {
+				map.value = "inactive"
+				map.descriptionText = "$device.displayName detected no motion"
+			} else if (cmd.event == 0x07 || cmd.event == 0x08) {
+				map.value = "active"
+				map.descriptionText = "$device.displayName detected motion"
+			}
+		}
+	} else if (cmd.hasProperty("sensorType")) {
+		componentLabel = "Qubino Temperature Sensor"
+		childDTHname = "Child Temperature Sensor"
+		switch (cmd.sensorType) {
+			case 0x01:
+				log.debug ("cmd: ${cmd}")
+				map.name = "temperature"
+				def cmdScale = cmd.scale == 1 ? "F" : "C"
+				map.value = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmdScale, cmd.precision)
+				map.unit = getTemperatureScale()
+				break
+		}
+	}
+	childDni = "${device.deviceNetworkId}-${childUniqueDni[componentLabel]}"
+
+	if (doesChildDeviceExist(componentLabel)) {

```

Comment:
I would move this section (searching / creating a child device) before creating a notification event - for better code readability.

Reply:
@KKlimczukS Actually this section has to be called after creating the notification event because the arguments for searching/creating functions are not specified before. I can just move that section outside.

Reply:
@BJanuszS ok, let's leave it where it is. You may just rename that whole method to handleChildEvent()